### PR TITLE
Protect: Add Footer to the admin page

### PR DESCRIPTION
**Pull request by leogermani**
#### Changes proposed in this Pull Request:

* Add Footer to the admin page

#### Jetpack product discussion

p1HpG7-fgl-p2

#### Does this pull request change what data or activity we track or use?

No

#### Testing instructions:

* Connect Protect
* Make sure you see the Footer and that it looks good
* Confirm that clicking the Learn more link takes you to the redirect service (the redirect is not registered yet because the page doesn't exist)
* Confirm that clicking the button takes you to the checkout page. 
* Verify the Checkout URL for the site url and redirect url
* verify that when the user is not connected `unlinked=1` is added to the URL

---
- To see the specific tasks where the Asana app for GitHub is being used, see below:
  - https://app.asana.com/0/0/1202183900595779

### CODE REVIEW

File: projects/plugins/protect/src/js/components/admin-page/index.jsx [hunk 1]
```diff
@@ -12,9 +20,22 @@ import { useConnection } from '@automattic/jetpack-connection';
...
+		'jetpack_security_t1_yearly',
+		siteSuffix,
+		adminUrl,
+		isUserConnected
+	);
+	const learnMoreUrl = getRedirectUrl( 'jetpack-protect-footer-learn-more' );
+
+	const handleProductButton = useCallback( () => {
+		window.location = securityCheckoutUrl;
+	}, [ securityCheckoutUrl ] );
```
Comment (retrofox) [inline projects/plugins/protect/src/js/components/admin-page/index.jsx:38]: sorry for asking this but I wonder whether we can land https://github.com/Automattic/jetpack/pull/24122 first and then update this section, with something like...

```es6
const { run } = useProductCheckoutWorkflow( {
	productSlug: SECURITY_BUNDLE,
	siteSuffix,
	redirectUrl,
} );
```
It continues below...

File: projects/plugins/protect/src/js/components/admin-page/index.jsx [hunk 2]
```diff
@@ -45,6 +66,7 @@ const Admin = () => {
 					</Col>
 				</Container>
 			</AdminSectionHero>
+			<Footer handleProductButton={ handleProductButton } learnMoreUrl={ learnMoreUrl } />
```
Comment (retrofox) [inline projects/plugins/protect/src/js/components/admin-page/index.jsx:69]: ... continues from above:

```jsx
<Footer handleProductButton={ run } learnMoreUrl={ learnMoreUrl } />
 ```

File: projects/plugins/protect/src/js/components/footer/index.jsx [hunk 3]
```diff
@@ -4,13 +4,14 @@
...
 import { __ } from '@wordpress/i18n';
+import PropTypes from 'prop-types';
 
 /**
  * Internal dependencies
  */
 import styles from './styles.module.scss';
 
-const Footer = () => {
+const Footer = ( { handleProductButton, learnMoreUrl } ) => {
```
Comment (retrofox) [inline projects/plugins/protect/src/js/components/footer/index.jsx:14]: Actually... I wonder whether we can make this component self sufficient, moving the checkout logic straight here. We can remove the props, too. That's a benefit about using react hooks.

File: projects/plugins/protect/src/js/components/admin-page/index.jsx (hunk 1 above)
Comment (leogermani) [reply projects/plugins/protect/src/js/components/admin-page/index.jsx:38]: Sure, I was expecting something was going to change after your PRs (I sent you a ping in Asana about it ;) )

File: projects/plugins/protect/src/js/components/footer/index.jsx (hunk 3 above)
Comment (leogermani) [reply projects/plugins/protect/src/js/components/footer/index.jsx:14]: I tried to leave this component as a pure layout component as possible... to be easier to extract it if we eventually decide to move it com js-packages/components as one reusable element

**Review by retrofox — APPROVED** [review]
LGTM.
follow-up tasks:
* expose siteSuffix and adminUrl in packages/connection
* Change the footer plan section when the user already has bought the plan

**Pull request merged**